src/components/Reports/PartnerGivingAnalysisReport/PartnerGivingAnalysisReport.tsx

```diff
@@ -61,8 +58,12 @@ export const PartnerGivingAnalysisReport: React.FC<Props> = ({
     [activeFilters, searchTerm],
   );
 
-  const { data, previousData, loading } = usePartnerGivingAnalysisQuery({
-    variables: {
+  // pageSize is intentionally omitted from the dependencies array so that the query isn't reloaded when the page size changes
+  // If all the pages have loaded and the user changes the page size, there's no reason to reload all the pages
+  const pageSize =
+    apiRef.current?.state.pagination.paginationModel.pageSize ?? null;
```

Comment:
My broken code, but apparently passing in `first: null` loads all rows instead of the default 25!

```suggestion
  const pageSize =
    apiRef.current?.state.pagination.paginationModel.pageSize;
```